patterns.go

```diff
@@ -30,6 +50,26 @@ var Reliable = Or(TCP, UTP, QUIC)
 // IPFS can run over any reliable underlying transport protocol
 var IPFS = And(Reliable, Base(ma.P_IPFS))
 
+// Define http over TCP or DNS or http over DNS format multiaddr
+var HTTP = Or(
+	And(TCP, Base(ma.P_HTTP)),
+	And(IP, Base(ma.P_HTTP)),
+	And(DNS, Base(ma.P_HTTP)),
+	And(DNS),
```

Comment:
Are you sure this is correct, this would mean that HTTP pattern will match just DNS addr.

Reply:
Right, this was carried over from the original PR, which in turn was carried over from js-mafmt: https://github.com/multiformats/js-mafmt/blob/master/src/index.js#L43.

Reported it there: https://github.com/multiformats/js-mafmt/issues/40#issuecomment-467909954.

I don't think the intention is to let a solitary dns-type addr imply HTTP by default, no?

Reply:
I agree with it not implying HTTP.